### ISSUE TITLE
Looking for a way to create cohort based on user creation time

### ISSUE THREAD

**Issue by swinster** [catalyst/moodle-local_cohortauto#12]
Hi,

I guess the field to reference is `{{ timecreated }}`, but then I'm not entirely sure how.

What is required is to create cohorts based on Month or bi-min they were created e.g.:

* Oct 2020, Nov 2020, Dec 2020
of
* Sept-Oct 2020, Nov-Dec 2002

etc

Some pointer would be awsome

**Comment by danmarsden** [comment]
It uses mustache templating so this page is probably a good place to look for ideas - I wonder if userdate might work:
https://docs.moodle.org/dev/Templates#.7B.7B.23_userdate_.7D.7D_.28Moodle_3.3_onwards.29

I don't think we've tried anything like this before so let us know if you figure it out!

**Comment by swinster** [comment]
Well, I used this pattern in the Main Template section, but nothing happened:

```
{{ #userdate }} {{ timecreated }}, %B %Y {{ /userdate }}
```

so I then tried adding spaces:

```
{{  #userdate }} {{ timecreated }}, %B %Y {{  /userdate }}
```

Which resulted in a cohort:

```
 1605290540, %B %Y 
```

Obviously, the epoch time and string literals. 

I looked at the "Replace array" but it is simply looks to use a static text pattern.

**Comment by swinster** [comment]
Does it look as if the plugin ignores helpers?

**Comment by danmarsden** [comment]
sounds like it - you might need to look to see how Moodle adds those helpers to the templates and see if you can hack that into the way the plugin manages templates.

**Comment by swinster** [comment]
I suspect this is beyond what I would be able to achieve :(. 

Just for a giggle, I might look at the code, but this is not really my area.

**Comment by swinster** [comment]
Yep - not giggling... Unfotuanley, I don't really know where to start.
